src/repo.rs

```diff
@@ -819,6 +819,54 @@ impl Repository {
         }
     }
 
+    /// Create a commit object from the given buffer and signature
+    ///
+    /// Given the unsigned commit object's contents, its signature and the
+    /// header field in which to store the signature, attach the signature to
+    /// the commit and write it into the given repository.
+    ///
+    /// Use `None` in `signature_field` to use the default of `gpgsig`, which is
+    /// almost certainly what you want.
+    ///
+    /// Returns the resulting (signed) commit id.
+    pub fn signed_commit(&self,
```

Comment:
Perhaps this could be called `commit_signed` to mirror the `commit` function above?